apps/src/templates/progress/progressHelpers.js

```diff
@@ -42,7 +37,34 @@ export function lessonIsVisible(lesson, state, viewAs) {
 }
 
 /**
- * Check to see if a stage/lesson is locked for all stages in the current section
+ * Treat the lesson as locked if either
+ * (a) it is locked for this user (in the case of a student)
+ * (b) non-verified teacher
+ * (c) signed out user
+ * @param {number} lesson - the lesson we're querying
+ * @param {object} state - State of our entire redux store
+ * @param {ViewType} viewAs - Are we interested in whether the lesson is viewable
+ *   for students or teachers
+ * @returns {boolean} True if the provided lesson is visible
+ */
+export function lessonIsLockedForUser(lesson, levels, state, viewAs) {
+  if (!lesson.lockable) {
+    return false;
+  }
+
+  if (!state.currentUser.userId) {
+    // Signed out user
+    return true;
+  } else if (viewAs === ViewType.Teacher) {
+    return !state.stageLock.lockableAuthorized;
+  } else if (viewAs === ViewType.Student) {
+    return stageLocked(levels);
+  }
+  return true;
```

Comment:
Just curious, why did you choose to default this to `true`?  (I'm not sure whether true or false is more appropriate so curious to hear your reasoning.)

Reply:
It just seemed like its always safer to lock the levels than unlock them. I don't think we should hit this case because signed out user, teacher and student should all be handled.